pkg/plugins/document/dynamodb/dynamodb.go

```diff
@@ -122,17 +124,49 @@ func (s *DynamoDocService) Delete(key *sdk.Key) error {
 		return fmt.Errorf("failed to marshal keys: %v", key)
 	}
 
-	input := &dynamodb.DeleteItemInput{
+	deleteInput := &dynamodb.DeleteItemInput{
 		Key:       attributeMap,
 		TableName: getTableName(*key.Collection),
 	}
 
-	_, err = s.client.DeleteItem(input)
+	_, err = s.client.DeleteItem(deleteInput)
 	if err != nil {
 		return fmt.Errorf("error deleting %v item %v : %v", key.Collection, key.Id, err)
 	}
 
-	// TODO: delete sub collection records
+	// Delete sub collection items
+	if key.Collection.Parent == nil {
+
+		queryInput := createDeleteQuery(key, nil)
+		resp, err := s.client.Query(queryInput)
+		if err != nil {
+			return fmt.Errorf("error performing delete: %v", err)
+		}
+
+		err = s.processDeleteQuery(key, resp)
+		if err != nil {
+			return fmt.Errorf("error performing delete: %v", err)
+		}
+
+		lastEvaluatedKey := resp.LastEvaluatedKey
+
+		for len(lastEvaluatedKey) > 0 {
```

Comment:
Could we reduce this duplicate delete query code using a recursive function instead? Or, alternatively put it in a while true loop, with a conditional 'break' statement once the lastEvaluatedKey is nil? Something like:

```
var lastEvaluatedKey = nil
for {
  queryInput := createDeleteQuery(key, lastEvaluatedKey)
  resp, err := s.client.Query(queryInput)
  if err != nil {
	  return fmt.Errorf("error performing delete: %v", err)
  }
  
  err = s.processDeleteQuery(key, resp)
  if err != nil {
	  return fmt.Errorf("error performing delete: %v", err)
  }
  
  lastEvaluatedKey = resp.LastEvaluatedKey
  if lastEvaluatedKey == nil {
    break
  }
}
```

Reply:
Sure will take a look

Reply:
Have done.